src/canopy-sofe-extensions.js

```diff
@@ -1,62 +1,72 @@
-          });
-        } else if (localRegex.test(localStorageValue)) {
-          postLocateNext(`${localDomain}:${localStorageValue}/${serviceName}.js`);
-        } else if (localStorageValue === 'stage') {
-          throw new Error(`'stage' is not a valid override for sofe services. Please change your sofe override for '${serviceName}' to be either 'app-stage' or 'cdn-stage'`);
-        } else {
-          postLocateNext(postLocateLoad);
-        }
+import { applyMiddleware, getManifest, getServiceName, getServiceUrl, InvalidServiceName } from 'sofe';
+
+const envs = {
+  integ: 'https://cdn-integ.canopy.ninja/sofe-manifest.json',
+  'cdn-integ': 'https://cdn-integ.canopy.ninja/sofe-manifest.json',
+  stage: 'https://cdn-stage.canopy.ninja/sofe-manifest.json',
+  'app-stage': 'https://cdn-stage.canopy.ninja/sofe-manifest.json',
+  'cdn-stage': 'https://cdn-stage.canopy.ninja/sofe-manifest.json',
+  prod: 'https://cdn.canopytax.com/sofe-manifest.json',
+};
+
+const localRegex = /^[0-9][0-9][0-9][0-9]$/
+
+const canopyMiddleware = () => (preLocateLoad, preLocateNext) => {
+  const serviceName = getServiceName(preLocateLoad);
+
+  preLocateNext(preLocateLoad);
+
+  return (postLocateLoad, postLocateNext) => {
+    if (serviceName) {
+      const localStorageValue = localStorage.getItem(`sofe:${serviceName}`);
+      if (typeof localStorageValue === 'string' && envs[localStorageValue]) {
+        getManifest(envs[localStorageValue])
+        .then(manifest => {
+          if (manifest[serviceName]) {
+            console.log(`Overriding sofe service '${serviceName}' to use the version found on '${localStorageValue}'`);
+            postLocateNext(manifest[serviceName]);
+          } else {
+            throw new Error(`Cannot find sofe service '${serviceName}' at the sofe manifest for '${localStorageValue}'`);
+          }
+        })
+        .catch(ex => {
+          throw ex;
+        });
+      } else if (localRegex.test(localStorageValue)) {
+        postLocateNext(`https://localhost:${localStorageValue}/${serviceName}.js`);
+      } else if (localStorageValue === 'stage') {
+        throw new Error(`'stage' is not a valid override for sofe services. Please change your sofe override for '${serviceName}' to be either 'app-stage' or 'cdn-stage'`);
       } else {
-        console.warn(`canopy-sofe-extensions not working properly (overrides to "cdn-stage", "thanos", etc) because sofe's getServiceName function returned something falsy for`, preLocateLoad);
         postLocateNext(postLocateLoad);
       }
+    } else {
+      console.warn(`canopy-sofe-extensions not working properly (overrides to "cdn-stage", "thanos", etc) because sofe's getServiceName function returned something falsy for`, preLocateLoad);
+      postLocateNext(postLocateLoad);
     }
   }
+}
+
+applyMiddleware(canopyMiddleware);
+
+export function calculatePublicPath(serviceName) {
+  let url
+
+  try {
+    url = getServiceUrl(serviceName)
+  } catch (err) {
```

Comment:
Do we want to put any error handling in this block?

Reply:
There are two different ways we try to get the url -- so if this one fails we move on to the next one. But if both fail, we throw [this error](https://github.com/CanopyTax/canopy-sofe-extensions/pull/7/files#diff-1493740359815eecbe0d0d9eea309af2R70)